src/game/behaviors/red_coin.inc.c

```diff
@@ -50,18 +50,34 @@ void bhv_red_coin_loop(void) {
             o->parentObj->oHiddenStarTriggerCounter++;
 
             // Spawn the orange number counter, as long as it isn't the last coin.
-            if (o->parentObj->oHiddenStarTriggerCounter != 8) {
-                spawn_orange_number(o->parentObj->oHiddenStarTriggerCounter, 0, 0, 0);
+            if (o->parentObj->oHiddenStarTriggerCounter != o->parentObj->oHiddenStarTriggerTotal) {
+                if (o->parentObj->oHiddenStarTriggerCounter > 99) {
```

Comment:
I think it'd be good to have a comment here like
```suggestion
                // Cap visible count to 99
                if (o->parentObj->oHiddenStarTriggerCounter > 99) {
```

just for clarity on how this works when going over 99